cloudsync/long_poll.py

```diff
@@ -97,3 +101,13 @@ def stop(self, forever=True, wait=False):
         # Don't wait for do() to finish, could wait for up to long_poll_timeout seconds
         self.unblock()
         super().stop(forever=forever, wait=wait)
+        if self.runnable_thread_id and self.runnable_thread_id in LONG_POLLERS:
```

Comment:
i would do this as a test-only wrapper class that tracks calls to start() stop(), that way 

a) applies to any runnable
b) no test-tracking-code in the main system